examples/beam_tutorial.py

```diff
@@ -0,0 +1,240 @@
+        room_length = params['room_length']
+
+        J['deflection','room_width'] = (-192.0 * E * I * TOTAL_LOAD_PSI) / ((5.0 * room_length**3) * (TOTAL_LOAD_PSI * room_width/2.0 + BEAM_WEIGHT_LBS_PER_IN)**2)
+        J['deflection','room_length'] = (-1152.0 * E * I) / (5.0 * ((TOTAL_LOAD_PSI * room_width)/2.0 + BEAM_WEIGHT_LBS_PER_IN) * room_length**4)
+
+        return J
+
+
+class BendingStressDiscipline(Component):
+    """Component containing Bending Discipline."""
+
+    def __init__(self):
+        super(BendingStressDiscipline, self).__init__()
+        
+        self.add_param('room_width', val=0.0)
+        self.add_param('room_length', val=0.0)
+        self.add_output('bending_stress_ratio', val=0.0)
+
+    def solve_nonlinear(self, params, unknowns, resids):        
+
+        room_width = params['room_width']
+        room_length = params['room_length']
+
+        unknowns['bending_stress_ratio'] = (0.5*BEAM_HEIGHT_IN * ((0.5 * TOTAL_LOAD_PSI * room_width)  + BEAM_WEIGHT_LBS_PER_IN) * (room_length)**2) / (8.0 * YIELD_STRENGTH_PSI * I)
+
+    def linearize(self, params, unknowns, resids):        
+        J = {}
+
+        room_width = params['room_width']
+        room_length = params['room_length']
+
+        J['bending_stress_ratio','room_width'] = (room_length**2) * BEAM_HEIGHT_IN * (TOTAL_LOAD_PSI*room_width/2.0 + BEAM_WEIGHT_LBS_PER_IN) / (16.0 * I * YIELD_STRENGTH_PSI)
+        J['bending_stress_ratio','room_length'] = (BEAM_WEIGHT_LBS_PER_IN + (TOTAL_LOAD_PSI*room_width/2.0)) * BEAM_HEIGHT_IN * room_length / (8.0 * I * YIELD_STRENGTH_PSI)
+
+        return J
+
+class ShearStressDiscipline(Component):
+    """Component containing Bending Discipline."""
+
+    def __init__(self):
+        super(ShearStressDiscipline, self).__init__()
+        
+        self.add_param('room_width', val=0.0)
+        self.add_param('room_length', val=0.0)
+        self.add_output('shear_stress_ratio', val=0.0)
+
+    def solve_nonlinear(self, params, unknowns, resids):        
+
+        room_width = params['room_width']
+        room_length = params['room_length']
+
+        unknowns['shear_stress_ratio'] = 0.5 * ((0.5 * TOTAL_LOAD_PSI * room_width)  + BEAM_WEIGHT_LBS_PER_IN) * (room_length) / (CROSS_SECTIONAL_AREA_SQIN * YIELD_STRENGTH_PSI)
+
+    def linearize(self, params, unknowns, resids):        
+        J = {}
+
+        room_width = params['room_width']
+        room_length = params['room_length']
+
+        J['shear_stress_ratio','room_width'] = TOTAL_LOAD_PSI * room_length / (4.0 * YIELD_STRENGTH_PSI * CROSS_SECTIONAL_AREA_SQIN)
+        J['shear_stress_ratio','room_length'] = (BEAM_WEIGHT_LBS_PER_IN + (TOTAL_LOAD_PSI * room_width / 2.0))/(2.0 * YIELD_STRENGTH_PSI * CROSS_SECTIONAL_AREA_SQIN)
+
+        return J
+
+
+class BeamTutorialDerivatives(Group):
+   
+    def __init__(self):
+        super(BeamTutorialDerivatives, self).__init__()
+
+        self.add('ivc_rl', IndepVarComp('room_length', 100.0), promotes=['*'])
```

Comment:
Avoid using promotes=['*'] in general, and especially in a tutorial, since a tutorial is supposed to show best practices.  Ideally, don't use promotes at all and instead use explicit connect statements.  If that's too cumbersome, then use promotes with lists of specific variable names and no wildcards.